### DISCUSSION TITLE
Could not find 'nx' module in this workspace

### DISCUSSION THREAD

**Discussion by wuarmin** [nrwl/nx#16104] [General]
Hey,

I followed this this post https://www.javierbrea.com/blog/pnpm-nx-monorepo-02/ and built a pnpm mono-repo, which uses nx. I have one app named `myapp` and two packages named `shared-ui` and `shared-features`.

To build `myapp` I have following nx-configuration:

```json
"targetDependencies": {
    "start": [
      {
        "target": "build",
        "projects": "dependencies"
      }
    ]
  }
``` 
I think nowadays, it could be changed to:
```json
  "targetDefaults": {
    "start": {
      "dependsOn": ["^build"]
    }
  }
```

Doesn't matter! Either way, I get the following error when I try to start myapp and use latest nx. The packages are built properly, but the `myapp` start fails.

f I call `pnpm nx start myapp`, I get following error:
```txt
dist/select-utils-14284a03.js              2.84 kB │ gzip: 1.06 kB
✓ built in 3.01s

> nx run myapp:start

Could not find 'nx' module in this workspace. Error: Invalid source or target versions. Source: 5.0, Target: 5.1.
Only backwards compatibility between "5.1" and "5.0" is supported.
This error can be caused by "@nrwl/..." packages getting out of sync or outdated project graph cache.
Check the versions running "nx report" and/or remove your "nxdeps.json" file (in node_modules/.cache/nx folder).
```

It seems the reason is, that `nx` is installed as a dependency of the workspace, and therefore `nx run myapp:start` get called without `pnpm` prefixed. Do you know, how I can fix this?

It worked with older nx-versions.

Thanks and best regards!

**Comment by stevebeauge** [comment]
Same issue. I workaround by installing nx again `pnpm -w add -D nx`. Don't know why, but it unlocks my local build process (never worked on Azure devops CI)

**Comment by moatorres** [comment]
Forgive me if I didn't understand your question, but I think there are a couple of things to consider. The first one (and you probably already know) is that we have to install `nx` globally to use it as a system package. If you prefer to have `nx` as a development dependency you can run `yarn nx run myapp:start` or `npm nx run myapp [...]` or `pnpm nx run [...]` or `pnpm exec nx run [...]`. The second thing is that your example seems to be missing an executor. 

I've pasted an example from one of my repos ([https//github.com/codyslexia/nexa](https//github.com/codyslexia/nexa)). It serves an Apollo GraphQL application. I can run it with `pnpm nx run graphql-users:serve` or even shorter `pnpm nx serve graphql-users`. 

```json
{
  "name":"graphql-users",
  "$schema":"../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot":"apps/graphql/users/src",
  "projectType":"application",
  "serve":{
    "executor":"@nx/js:node",
    "defaultConfiguration":"development",
    "options":{
      "buildTarget":"graphql-users:build"
    },
    "configurations":{
      "development":{
        "buildTarget":"graphql-users:build:development"
      },
      "production":{
        "buildTarget":"graphql-users:build:production"
      }
    }
  }
}
```
